### PR TITLE
Improvement/navigation/profile

### PR DESCRIPTION
**Pull request by julia-andersson**
## [APP-XXX]

- Profile navigation

## Checklist

- [ ] Dark/light mode verification
- [ ] Landscape verification
- [ ] iPad verification
- [ ] iPod/iPhone 12 mini/iPhone SE verification

### CODE REVIEW

File: Projects/Profile/Sources/Views/Screens/EuroBonus/ChangeEuroBonusView.swift [hunk 1]
```diff
@@ -42,37 +48,13 @@ private class ChangeEurobonusViewModel: ObservableObject {
...
-                    }
-            }
-        }
-        .configureTitle(L10n.SasIntegration.enterYourNumber)
-        .onAction(ProfileStore.self) { action in
-            if case .dismissChangeEuroBonus = action {
-                PopJourney()
-            }
+            /* TODO: SOMEHOW SET SUCESS HERE*/
+            //  router.push(EuroBonusRouterType.successChangeEuroBonus)
```
Comment (julia-andersson) [inline Projects/Profile/Sources/Views/Screens/EuroBonus/ChangeEuroBonusView.swift:53]: @sladan-hedvig having some issues here on how to push success screen. Any ideas?

File: Projects/Profile/Sources/Navigation/ProfileNavigation.swift [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        .environmentObject(profileNavigationViewModel)
+        .detent(
+            item: $profileNavigationViewModel.isDeleteAccountPresented,
+            style: .height,
+            options: .constant(.withoutGrabber)
+        ) { memberDetails in
+            redirect(
+                .deleteAccount(
+                    memberDetails: memberDetails
+                        //                onDismiss: {
```
Comment (julia-andersson) [inline Projects/Profile/Sources/Navigation/ProfileNavigation.swift:56]: @sladan-hedvig this is where I tried to add dismiss closure to deleteAccount but it doesn't work. We need this to set profileNavigationViewModel.isDeleteAccountPresented = nil. Any idea on ay other approach?

case deleteAccount(memberDetails: MemberDetails, onDismiss: () -> Void)

**Comment by julia-andersson** [comment]
Having issues with how to embed navigation. Looks like this atm. 

simulator_screenshot_BFECFEFC-11BD-489B-91D6-3A7AE545C18C

File: Projects/Profile/Sources/Navigation/ProfileNavigation.swift (hunk 2 above)
Comment (sladan-hedvig) [reply Projects/Profile/Sources/Navigation/ProfileNavigation.swift:59]: I pushed commit that should resolve this issue.
Basically having control on the profileNavigationViewModel from the both main and profile navigation.
Please give it a look. Something simular we have with the home vm

**Comment by sladan-hedvig** [comment]
> @sladan-hedvig not sure on how to handle push notifications with the new navigation.

I have some ideas but lets leave it for the next week.

**Comment by sladan-hedvig** [comment]
> Having issues with how to embed navigation. Looks like this atm.
> 
> simulator_screenshot_BFECFEFC-11BD-489B-91D6-3A7AE545C18C

Yeah, you had navigation inside navigation. I just removed eurobonus router

**Comment by sladan-hedvig** [comment]
> @sladan-hedvig not sure on how to handle push notifications with the new navigation.

Lets have a meeting for this to see how we are going to handle it

File: Projects/App/Sources/MainNavigationJourney.swift [hunk 3]
```diff
@@ -267,14 +267,74 @@ struct MainNavigationJourney: App {
...
+                } onCancel: {
+                    router.dismiss()
+                }
+            case .deleteRequestLoading:
+                DeleteRequestLoadingView(
+                    screenState: .success,
+                    dismissAction: { profileDismissAction in
+                        switch profileDismissAction {
+                        case .makeHomeTabActiveAndOpenChat:
+                            vm.selectedTab = 0
```
[the remark this answers is not in the document]
Comment (sladan-hedvig) [reply Projects/App/Sources/MainNavigationJourney.swift:328]: This should be fine now I think. Haven't we fixed this?

**Review by sladan-hedvig — APPROVED** [review]
Just don't forget to uncomment this:
if store.state.showTravelCertificate
if store.state.partnerData?.shouldShowEuroBonus

🥳

**Pull request merged**